src/hooks.c

```diff
@@ -514,13 +514,20 @@ pathman_rel_pathlist_hook(PlannerInfo *root,
 	rel->partial_pathlist = NIL;
 #endif
 
+/* Convert list to array for faster lookups */
+#if PG_VERSION_NUM >= 110000
+	setup_append_rel_array(root);
```

Comment:
Are you sure this is ok? We might add new `AppendRelInfos` each time `pathman_rel_pathlist_hook` is called.

Reply:
Yeah, this is not good. Fixed.